corehq/apps/enterprise/api/keyset_paginator.py

```diff
@@ -0,0 +1,94 @@
+from django.http.request import QueryDict
+from urllib.parse import urlencode
+from tastypie.paginator import Paginator
+
+
+class KeysetPaginator(Paginator):
+    '''
+    An alternate paginator meant to support paginating by keyset rather than by index/offset.
+    `objects` is expected to represent a query object that exposes an `.execute(limit)`
```

Comment:
Why not having a base `QueryObject` class, and having a function `execute` that will raise `NotImplemented` error. Then have a docstring to explain what do you expect from the `execute` function or even give an example. Anyone who wants to use this `KeysetPaginator` should pass a object inherits from the base `QueryObject`